src/typeInference/annotateBinaryOperator.ts

```diff
@@ -40,7 +40,7 @@ export function annotateBinOp(
 	if (valueType) {
 		const type = { type: valueType, mult: left.mult };
 
-		if (valueType !== ValueType.XSNUMERIC) {
+		if (valueType !== ValueType.XSNUMERIC && valueType !== ValueType.ITEM) {
```

Comment:
is there a particular reason for this?

Reply:
Yes, we don't want to annotate the AST with item()*